accepted/0000-yomi-initial-integration.md

```diff
@@ -0,0 +1,488 @@
+  installation.
+
+- yomi-storage: device partition, LVM or RAID configuration, and file
+  systems applied.
+
+- yomi-users: creation of initial users, passwords and certificates.
+
+As we will see in future sections, those forms contains some
+overlapping with others Salt formulas (like the locale configuration
+that `systemd-firstboot` requires), and with other subsystems in Uyuni
+(like the software section).
+
+## Highstate and Workflow
+
+Installing a new system will be equivalent of applying a highstate to
+a system booted from the Yomi image.
+
+The current Yomi image is a JeOS based image with a `salt-minion`
+service enabled, some systemd overwrites that will help during the
+initial configuration, and the CLI tools required when a new
+installation is performed. You can find more information here:
+https://github.com/openSUSE/yomi#the-iso-image
+
+The current Yomi states are aware of differentiating when the system
+is booting from an external image (like the Yomi image), from when the
+boot comes from the local device. This mechanism is used to decide
+when the installation of a system makes sense or not when a highstate
+is re-applied.
+
+This means that we can set all the pillar data for the system, without
+risk of breaking or damaging an already installed system.
+
+This open the question about the workflow that will be in place when a
+new system is allocated. The current prototype was tested in this way:
+
+1) Boot the new system via the Yomi ISO image. The local DHCP server
+   assign a FQDN to the system, and this is used automatically as a
+   minion ID.
+
+   (note: the master address was indicated via a GRUB command line
+   that a systemd service overwrite in the image can read and
+   interpret)
+
+2) Accept the Salt key in the manager (Uyuni 4.0.2), and register the
+   new discovered system.
+
+3) Assign to the new system all the Yomi states (Installer group),
+   fulfill the forms, and apply the highstate to the new node.
+
+Once the new system reboot, the highstate can be re-applied without
+consequences, as the state detect early that the root filesystem
+device is already mounted, and no subsequent states from Yomi will be
+executed.
+
+Some minor bugs in Uyuni (bsc#1150586, bsc#1150589, bsc#1150590,
+bsc#1152673, bsc#1152767) makes this process less direct, but for all
+except bsc#1152673 Yomi is providing a workaround. The remaining bug
+requires some manual intervention before the highstate can be applied
+from Uyuni.
+
+The big problem for this workflow has been pointed by Joachim Werner,
+and is about the default values for the pillar data. IIUC the problem
+is that if Yomi provides some default pillar configuration, the new
+discovered node can wrongly apply a highstate (that AFAIK are applied
+automatically or scheduled by the user) with non-optimal data (disk
+configuration, packages, repositories, etc)
+
+There are two proposals, also from Joe, to address this issue (copied
```

Comment:
As the spirit of an RFC is to submit a design proposal to the team in order to gather feedback and ultimately agree on a text that can be used as a guideline for the next implementation step.

I would thus suggest taking a decision here (and in all other cases where multiple alternatives are possible) and propose just one direction. The pros/cons wrt the other known possibilities should be described separately in the "Alternatives" section below.

Also note that an RFC doesn't necessarily need to be an all-encompassing design document over a big solutions with many unknowns still to discover - setting a general design direction and specifying "just enough" so that the first iteration can be implemented is sufficient. What I often see is a separate "Next Steps" paragraph, which is informative but not really binding to the acceptance of the RFC, or even subsequent RFCs (once the first "chunk" is implemented/feedback is gathered/next step is clearer).

Reply:
This is what I try do to in the areas that I have the knowledge to make an informed decisions. The places with open questions are to motivate the discussion.

But I will take a guess in the RFC and go for a single proposal.

Reply:
Done. I moved this part to the unresolved question, and I moved to a workflow that I think is more similar to the first one proposed.